src/__tests__/tsconfig-loader.test.ts

```diff
@@ -167,262 +164,135 @@ describe("walkForTsConfig", () => {
-    const actualConfig = {
-      extends: [
-        "./base-config-1.json",
-        "./dir1/base-config-2.json",
-        "./dir1/dir2/base-config-3.json",
-      ],
-    };
-    const actualConfigPath = join("/root", "tsconfig.json");
-
-    const res = loadTsconfig(
-      join("/root", "tsconfig.json"),
-      (path) =>
-        [
-          baseConfig1Path,
-          baseConfig2Path,
-          baseConfig3Path,
-          actualConfigPath,
-        ].indexOf(path) >= 0,
-      (path) => {
-        if (path === baseConfig1Path) {
-          return JSON.stringify(baseConfig1);
-        }
-        if (path === baseConfig2Path) {
-          return JSON.stringify(baseConfig2);
-        }
-        if (path === baseConfig3Path) {
-          return JSON.stringify(baseConfig3);
-        }
-        if (path === actualConfigPath) {
-          return JSON.stringify(actualConfig);
-        }
-        return "";
-      }
-    );
+    const cwd = resolve(__dirname, "../../example/extend-multiple")
 
-    expect(res).toEqual({
-      extends: [
-        "./base-config-1.json",
-        "./dir1/base-config-2.json",
-        "./dir1/dir2/base-config-3.json",
-      ],
-      compilerOptions: {
-        baseUrl: join("dir1", "dir2"),
-        paths: { foo: ["bar2"] },
-      },
+    const result = tsConfigLoader({ cwd, getEnv: () => undefined });
+
+    expect(result).toEqual({
+      baseUrl: "./dir1/dir2",
+      paths: { foo: ["bar2"] },
+      tsConfigPath: resolve(cwd, "tsconfig.json")
     });
   });
 
   it("should load a config with array extends without .json extension", () => {
-    const baseConfig = {
-      compilerOptions: { baseUrl: ".", paths: { foo: ["bar"] } },
-    };
-    const baseConfigPath = join("/root", "base-config-1.json");
-    const actualConfig = { extends: ["./base-config-1"] };
-    const actualConfigPath = join("/root", "tsconfig.json");
-
-    const res = loadTsconfig(
-      join("/root", "tsconfig.json"),
-      (path) => [baseConfigPath, actualConfigPath].indexOf(path) >= 0,
-      (path) => {
-        if (path === baseConfigPath) {
-          return JSON.stringify(baseConfig);
-        }
-        if (path === actualConfigPath) {
-          return JSON.stringify(actualConfig);
+    const cwd = resolve(__dirname, "../../example/extend-without-extension")
+
+    const result = tsConfigLoader({ cwd, getEnv: () => undefined });
+
+    expect(result).toEqual({
+      baseUrl: "./",
+      paths: { foo: ["bar"] },
+      tsConfigPath: resolve(cwd, "tsconfig.json")
+    });
+  });
+
+  it("should resolve multiple levels of tsconfig extension", () => {
+    const cwd = resolve(__dirname, "../../example/inherited");
+    const result = tsConfigLoader({ cwd, getEnv: () => undefined });
+
+    expect(result).toEqual({
+      baseUrl: undefined,
+      paths: { "@": [] },
+      tsConfigPath: resolve(cwd, "tsconfig.json"),
+    });
+  });
+});
+
+describe("getTsconfig", () => {
```

Comment:
I extracted these tests which only target the `getTsconfig` function from `get-tsconfig` to show that other details like `strict` are correctly inherited (since this information is not returned from `loadSyncDefault`)